app/models/user.rb

```diff
@@ -104,6 +104,39 @@ def self.portus
     find_by(username: "portus")
   end
 
+  # Creates a user and sets this password as an empty string. This way we don't
+  # store sensible data that is already stored in another place (e.g. LDAP), and
+  # regular login won't be allowed for such users. The `params` argument is the
+  # parameters that will be passed into the `User.create` method call.
+  #
+  # Returns two objects: a user object, and a boolean stating whether everything
+  # was ok or not. Callers should check this second object before doing anything
+  # at all with the first (since it might be an invalid user object).
+  def self.create_without_password(params)
+    user    = nil
+    created = false
+
+    ActiveRecord::Base.transaction do
+      # We create it first with a dummy password that will be erased later on.
+      user = User.create(params.merge(password: "12341234"))
```

Comment:
Can we generate a random string just to avoid this hardcoded string? Maybe with SecureRandom `alphanumeric` or `hex`.

Reply:
Done!